src/hooks/useHttp.test.js

```diff
@@ -0,0 +1,55 @@
+import { act, renderHook } from '@testing-library/react-hooks';
+import { fetchVideos } from '../lib/api';
+import useHttp from './useHttp';
+import mockData from '../helper/mock-data.json';
+
+describe('useHttp Hook', () => {
+  window.fetch = jest.fn();
+
+  test('fetches videos by url constructed correcty', async () => {
+    const url = `https://www.googleapis.com/youtube/v3/search?part=snippet&key=AIzaSyBfc4oeDp-VCHubNeaFbwuPz-_LKiWMn7E&maxResults=20&q=react&type=video&safeSearch=strict`;
```

Comment:
**IMPORTANT**:

It is dangerous to upload your API key to github.  I recommend you use the `.env` package so you manage them using variables that you never upload, you will need to add them to Netlify as well so they work on the server but they won't be publicly available as they are now.

Take a look at [Adding Custom Environment Variables](https://create-react-app.dev/docs/adding-custom-environment-variables/)

Reply:
Well, this was a mistake that I had, I could put anything instead of the API KEY, since it was to simulate the url, but I missed it. Thanks also.